applications/HDF5Application/custom_io/hdf5_connectivities_data.cpp

```diff
@@ -1,257 +1,207 @@
-    SetData(name, rElements);
+    KRATOS_ERROR_IF_NOT(mpFile->HasPath(entity_group_path))
+        << "Entity data for \"" << rEntityName << "\" not found in file [ expected path in file = "
+        << entity_group_path << " ].\n";
 
-    KRATOS_CATCH("");
-}
+    IndexType start_index, block_size;
+    std::tie(start_index, block_size) = StartIndexAndBlockSize(*mpFile, entity_group_path);
 
-void ConnectivitiesData::SetData(const std::string& rName, ElementsContainerType const& rElements)
-{
-    KRATOS_TRY;
+    Vector<int> entity_ids, entitiy_property_ids;
+    Matrix<int> connectivities;
 
-    KRATOS_ERROR_IF_NOT(KratosComponents<ElementType>::Has(rName))
-        << "Name \"" << rName << "\" is not registered as an element." << std::endl;
-
-    if (rElements.empty())
-    {
-        Clear();
-        // Correctly set the name and array sizes. Otherwise, MPI collective write
-        // can fail due to inconsistent data across processes.
-        mName = rName;
-        const Element& r_elem = KratosComponents<ElementType>::Get(mName);
-        mConnectivities.resize(0, r_elem.GetGeometry().size());
-        return;
-    }
+    std::string file_entity_name;
+    mpFile->ReadAttribute(entity_group_path, "Name", file_entity_name);
 
-    CompareElementsAndConditionsUtility::GetRegisteredName(rElements.front(), mName);
-    KRATOS_ERROR_IF(mName != rName) << "Name \"" << rName << "\" is not the same as \"" << mName << "\"." << std::endl;
-    const int num_elems = rElements.size();
-    mIds.resize(num_elems, false);
-    mPropertiesIds.resize(num_elems, false);
-    const unsigned geometry_size = rElements.front().GetGeometry().size();
-    mConnectivities.resize(num_elems, geometry_size, false);
+    KRATOS_ERROR_IF_NOT(file_entity_name == rEntityName)
+        << "Entity name mismatch [ expected entity name = "
+        << rEntityName << ", entity name in file = " << file_entity_name << " ].\n";
 
-    // Fill arrays and perform checks.
-    #pragma omp parallel for
-    for (int i = 0; i < num_elems; ++i)
-    {
-        ElementsContainerType::const_iterator it = rElements.begin() + i;
-        // Check that the element and geometry types are the same.
-        KRATOS_ERROR_IF_NOT(GeometricalObject::IsSame(*it, rElements.front()))
-            << "Element #" << it->Id() << " is not the same as #" << rElements.front().Id() << '!' << std::endl;
-        // Fill ids.
-        mIds[i] = it->Id();
-        mPropertiesIds[i] = it->GetProperties().Id();
-        const ElementType::GeometryType& r_geom = it->GetGeometry();
-        // Fill connectivities.
-        for (unsigned k = 0; k < geometry_size; ++k)
-            mConnectivities(i, k) = r_geom[k].Id();
-    }
+    mpFile->ReadDataSet(entity_group_path + "/Ids", entity_ids, start_index, block_size);
+    mpFile->ReadDataSet(entity_group_path + "/PropertiesIds", entitiy_property_ids, start_index, block_size);
```

Comment:
```suggestion
    mpFile->ReadDataSet(entity_group_path + "/PropertyIds", entitiy_property_ids, start_index, block_size);
```

how big of a change would this entail? Duplicate plurals in kratos names are driving me mad.

Reply:
I would say, we would be breaking all the old HDF5 files. Let's do it in a separate PR then.